### PR TITLE
replace should.be with should.be.same.as, make should.be a placeholder for should.equal.

### PR DESCRIPTION
**Pull request by FeepingCreature**
ping @linkrope

### CODE REVIEW

File: src/dshould/basic.d [hunk 1]
```diff
@@ -48,25 +94,23 @@ if (isInstanceOf!(ShouldType, Should) && !should.hasWord!"approximately")
             {
                 check(
                     got !is expected,
-                    format("%s%s", refInfo, expected.quote),
-                    isReferenceType ? "same reference" : "it",
+                    format("%s%s", "different reference than ", expected.quote),
```
Comment (linkrope) [inline src/dshould/basic.d]: format!"different reference than %s"(expected.quote)

File: src/dshould/basic.d [hunk 2]
```diff
@@ -48,25 +94,23 @@ if (isInstanceOf!(ShouldType, Should) && !should.hasWord!"approximately")
...
+            static if (isNullType)
             {
                 check(got is null, "null", format("%s", got.quote), file, line);
             }
             else
             {
                 check(
                     got is expected,
-                    format("%s%s", refInfo, expected.quote),
+                    format("%s%s", "same reference as ", expected.quote),
```
Comment (linkrope) [inline src/dshould/basic.d]: format!"same reference as %s"(expected.quote)

File: src/dshould/package.d [hunk 3]
```diff
@@ -136,11 +103,11 @@ unittest
     obj.should.be(null)
         .should.throwA!FluentException("Test failed: expected null, but got object.Object");
 
-    obj.should.not.be(obj)
+    obj.should.not.be.same.as(obj)
```
Comment (linkrope) [inline src/dshould/package.d:106]: move to `basic`, where `same` is defined?

File: src/dshould/basic.d (hunk 2 above)
Comment (FeepingCreature) [reply src/dshould/basic.d]: all replaced

File: src/dshould/basic.d (hunk 1 above)
Comment (FeepingCreature) [reply src/dshould/basic.d]: all replaced

File: src/dshould/package.d (hunk 3 above)
Comment (FeepingCreature) [reply src/dshould/package.d:106]: This tests the exception, not the comparison. The comparison is already tested in `basic`.

**Review by linkrope — APPROVED** [review]
update README?

**Comment by FeepingCreature** [comment]
done

**Pull request merged**